### ISSUE TITLE
[Wallet] "Unknown" panic on wallet 

### ISSUE THREAD

**Issue by mariocao** [witnet/witnet-rust#1747]
Some users have reported a `panic` while running the wallet.

```
thread '<unnamed>' panicked at 'no Task is currently running', C:\Users\IEUser\.cargo\registry\src\github.com-1ecc6299db9ec823\futures-0.1.30\src\task_impl\mod.rs:47:9
```
```
[2020-12-02T09:19:11Z INFO  parity_ws] Listening for new connections on 127.0.0.1:11212.
[2020-12-02T09:19:11Z INFO  witnet_net::client::tcp::actors::jsonrpc] Client "blocks" subscription created
[2020-12-02T09:19:11Z INFO  witnet_net::client::tcp::actors::jsonrpc] Client "superblocks" subscription created
[2020-12-02T09:19:11Z INFO  parity_ws::io] Accepted a new tcp connection from 127.0.0.1:56412.
[2020-12-02T09:19:32Z INFO  parity_ws::io] Accepted a new tcp connection from 127.0.0.1:56504.
[2020-12-02T09:20:38Z INFO  parity_ws::io] Accepted a new tcp connection from 127.0.0.1:56776.
[2020-12-02T09:20:46Z INFO  witnet_wallet::actors::app::routes] New WS notifications subscriber for session f58bc...
[2020-12-02T09:20:46Z INFO  witnet_wallet::actors::worker::methods] [SU] Starting synchronization of wallet 1923e947eefe8524ee3374446a0629cd523492b5f8170d8fef276a10c48416b9.
    	[Local beacon] CheckpointBeacon { checkpoint: 175729, hash_prev_block: 479d80a86de653bcb9f6f2a3ac695d321ae09e868e759cf98a1a177075f77766 }
    	[Node height ]   BlockInfo { checkpoint: 175780, block_hash: f366e9209f9ae9a0d5e7a861f65d454a2e9b53e149fefbc73c3b4c4fe94bbe3a }
thread '<unnamed>' panicked at 'no Task is currently running', /Users/user/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.30/src/task_impl/mod.rs:47:9
stack backtrace:
   0: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
   1: core::fmt::write
   2: std::io::Write::write_fmt
   3: std::panicking::default_hook::{{closure}}
   4: std::panicking::default_hook
   5: std::panicking::rust_panic_with_hook
   6: std::panicking::begin_panic
   7: futures::task_impl::current
   8: actix::address::channel::AddressSender<A>::send
   9: actix::address::Addr<A>::send
  10: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  11: futures_executor::local_pool::block_on
  12: witnet_wallet::actors::worker::methods::<impl witnet_wallet::actors::worker::Worker>::get_vt_outputs_from_pointers
  13: <core::iter::adapters::ResultShunt<I,E> as core::iter::traits::iterator::Iterator>::next
  14: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter
  15: core::iter::adapters::process_results
  16: witnet_wallet::actors::worker::methods::<impl witnet_wallet::actors::worker::Worker>::extend_transactions_data
  17: witnet_wallet::actors::worker::methods::<impl witnet_wallet::actors::worker::Worker>::index_block
  18: witnet_wallet::actors::worker::methods::<impl witnet_wallet::actors::worker::Worker>::handle_block
  19: witnet_wallet::actors::worker::methods::<impl witnet_wallet::actors::worker::Worker>::sync_inner
  20: witnet_wallet::actors::worker::methods::<impl witnet_wallet::actors::worker::Worker>::sync
  21: <actix::sync::SyncContextEnvelope<A,M> as actix::address::envelope::EnvelopeProxy>::handle
  22: actix::sync::SyncContext<A>::run
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
[2020-12-02T09:21:05Z WARN  witnet_wallet::actors::worker::methods] Tried to process a block #175781 that does not build directly on top our local (#175729) or confirmed tip (#175729)
```
See issue #1088 for a similar issue happening in the node.

Further investigation is needed in order to reproduce the problem and identify the bug.

**Comment by tmpolaczyk** [comment]
Same issue here, but different backtrace:

```
$ RUST_BACKTRACE=full RUST_LOG="witnet=debug" cargo run -- -c witnet_01.toml wallet server
    Finished dev [unoptimized + debuginfo] target(s) in 0.24s
     Running `target/debug/witnet -c witnet_01.toml wallet server`
Loading config from: witnet_01.toml
Setting log level to: DEBUG, source: Env
[2020-12-02T13:42:11Z DEBUG witnet_data_structures] Set environment to development
[2020-12-02T13:42:11Z INFO  witnet_net::client::tcp::actors::jsonrpc] Connecting client to 127.0.0.1:1234
[2020-12-02T13:42:11Z INFO  witnet_net::client::tcp::actors::jsonrpc] TCP socket is now connected to 127.0.0.1:1234
Error: IO error: While lock file: /home/tomasz/.local/share/witnet/witnet_wallet.db/LOCK: Resource temporarily unavailable
[2020-12-02T13:42:11Z DEBUG witnet_net::client::tcp::actors::jsonrpc] JsonRpcClient actor started!
[2020-12-02T13:42:11Z INFO  witnet_net::client::tcp::actors::jsonrpc] JsonRpcClient actor was trying to stop for some reason. Refusing to stop!
thread 'main' panicked at 'no Task is currently running', /home/tomasz/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.30/src/task_impl/mod.rs:47:9
stack backtrace:
   0:     0x55db9ea58a70 - std::backtrace_rs::backtrace::libunwind::trace::h72c2fb8038f1bbee
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/../../backtrace/src/backtrace/libunwind.rs:96
   1:     0x55db9ea58a70 - std::backtrace_rs::backtrace::trace_unsynchronized::h1e3b084883f1e78c
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/../../backtrace/src/backtrace/mod.rs:66
   2:     0x55db9ea58a70 - std::sys_common::backtrace::_print_fmt::h3bf6a7ebf7f0394a
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/sys_common/backtrace.rs:79
   3:     0x55db9ea58a70 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h2e8cb764b7fe02e7
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/sys_common/backtrace.rs:58
   4:     0x55db9ea7f9ec - core::fmt::write::h7a1184eaee6a8644
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/core/src/fmt/mod.rs:1080
   5:     0x55db9ea503e2 - std::io::Write::write_fmt::haeeb374d93a67eac
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/io/mod.rs:1516
   6:     0x55db9ea5ae9d - std::sys_common::backtrace::_print::h1d14a7f6ad632dc8
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/sys_common/backtrace.rs:61
   7:     0x55db9ea5ae9d - std::sys_common::backtrace::print::h301abac8bb2e3e81
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/sys_common/backtrace.rs:48
   8:     0x55db9ea5ae9d - std::panicking::default_hook::{{closure}}::hde0cb80358a6920a
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/panicking.rs:208
   9:     0x55db9ea5ab48 - std::panicking::default_hook::h9b1a691049a0ec8f
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/panicking.rs:227
  10:     0x55db9ea5b581 - std::panicking::rust_panic_with_hook::h2bdec87b60580584
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/panicking.rs:577
  11:     0x55db9d840d9c - std::panicking::begin_panic::{{closure}}::h32157e12cee84212
                               at /home/tomasz/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/panicking.rs:506
  12:     0x55db9d8421a9 - std::sys_common::backtrace::__rust_end_short_backtrace::h82fe09b2c717118a
                               at /home/tomasz/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/sys_common/backtrace.rs:153
  13:     0x55db9d840cc7 - std::panicking::begin_panic::h86ca1cd5774e449b
                               at /home/tomasz/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/panicking.rs:505
  14:     0x55db9d83b1dd - futures::task_impl::with::h890fa3b8ec50826b
                               at /home/tomasz/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.30/src/task_impl/mod.rs:47
  15:     0x55db9d83b332 - futures::task_impl::current::hb2f017f2a1babb0e
                               at /home/tomasz/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.30/src/task_impl/mod.rs:118
  16:     0x55db9c839f3c - actix::address::channel::AddressReceiver<A>::try_park::h1df070fe62928fee
                               at /home/tomasz/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-0.8.3/src/address/channel.rs:760
  17:     0x55db9c83ab8c - <actix::address::channel::AddressReceiver<A> as futures::stream::Stream>::poll::heb2d54f598875114
                               at /home/tomasz/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-0.8.3/src/address/channel.rs:798
  18:     0x55db9c831557 - actix::mailbox::Mailbox<A>::poll::hec038b5b3e96656f
                               at /home/tomasz/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-0.8.3/src/mailbox.rs:91
  19:     0x55db9c86f975 - <actix::contextimpl::ContextFut<A,C> as futures::future::Future>::poll::hbdf4ccaf2c66e290
                               at /home/tomasz/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-0.8.3/src/contextimpl.rs:364
  20:     0x55db9c86bf5b - <actix::contextimpl::ContextFut<A,C> as core::ops::drop::Drop>::drop::h6918700e9069d084
                               at /home/tomasz/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-0.8.3/src/contextimpl.rs:220
  21:     0x55db9c865576 - core::ptr::drop_in_place::h8002286c052b51b9
                               at /home/tomasz/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/mod.rs:175
  22:     0x55db9d81898c - core::ptr::drop_in_place::h94fec7e555b05901
                               at /home/tomasz/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/mod.rs:175
  23:     0x55db9d75b26e - core::ptr::drop_in_place::h4d6087887a6c29df
                               at /home/tomasz/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/mod.rs:175
  24:     0x55db9d78d495 - <alloc::vec::Vec<T> as core::ops::drop::Drop>::drop::h47f51a70e48ea64a
                               at /home/tomasz/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec.rs:2636
  25:     0x55db9d75ab66 - core::ptr::drop_in_place::h149890211498b8f1
                               at /home/tomasz/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/mod.rs:175
  26:     0x55db9d75c07f - core::ptr::drop_in_place::hba712c35ed6df288
                               at /home/tomasz/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/mod.rs:175
  27:     0x55db9d75ba56 - core::ptr::drop_in_place::h8cb4208b7b077233
                               at /home/tomasz/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/mod.rs:175
  28:     0x55db9d75b2eb - core::ptr::drop_in_place::h4de36324ff17c584
                               at /home/tomasz/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/mod.rs:175
  29:     0x55db9d77ffa7 - core::mem::drop::h3f5e78ba6b26c453
                               at /home/tomasz/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/mem/mod.rs:881
  30:     0x55db9d78afba - std::thread::local::fast::destroy_value::h8f9275510c64637f
                               at /home/tomasz/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/thread/local.rs:519
  31:     0x7f5e6bcef51f - __call_tls_dtors
  32:     0x7f5e6bceee8f - <unknown>
  33:     0x7f5e6bceeeba - exit
  34:     0x55db9ea5e667 - std::sys::unix::os::exit::h41e35182659f32e0
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/sys/unix/os.rs:608
  35:     0x55db9ea56e9f - std::process::exit::hc647caf088108c7c
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/process.rs:1713
  36:     0x55db9c1dacad - witnet::main::h70c773b8e8d7e074
                               at /home/tomasz/projects/witnet-rust/src/main.rs:18
  37:     0x55db9c21448b - core::ops::function::FnOnce::call_once::h59d17da3e8434048
                               at /home/tomasz/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:227
  38:     0x55db9c1cef1e - std::sys_common::backtrace::__rust_begin_short_backtrace::h28180af4dfbcc4be
                               at /home/tomasz/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/sys_common/backtrace.rs:137
  39:     0x55db9c0c96a1 - std::rt::lang_start::{{closure}}::h98f577da27c28d4b
                               at /home/tomasz/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/rt.rs:66
  40:     0x55db9ea5b9a7 - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::he179d32a5d10d957
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/core/src/ops/function.rs:259
  41:     0x55db9ea5b9a7 - std::panicking::try::do_call::hcb3d5e7be089b2b4
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/panicking.rs:381
  42:     0x55db9ea5b9a7 - std::panicking::try::h7ac93b0cd56fb701
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/panicking.rs:345
  43:     0x55db9ea5b9a7 - std::panic::catch_unwind::h7b40e396c93a4fcd
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/panic.rs:382
  44:     0x55db9ea5b9a7 - std::rt::lang_start_internal::h142b9cc66267fea1
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/rt.rs:51
  45:     0x55db9c0c9677 - std::rt::lang_start::he50cdb60329b1e4b
                               at /home/tomasz/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/rt.rs:65
  46:     0x55db9c1dadda - main
  47:     0x7f5e6bcd909b - __libc_start_main
  48:     0x55db9c0bf02a - _start
  49:                0x0 - <unknown>
```

**Comment by tmpolaczyk** [comment]
Should be fixed by #1759
